frontend/src/component/releases/ReleasePlanTemplate/MilestoneStrategyType.tsx

```diff
@@ -0,0 +1,59 @@
+import type { IFormErrors } from 'hooks/useFormErrors';
+import type { IReleasePlanMilestoneStrategy } from 'interfaces/releasePlans';
+import type { IStrategy } from 'interfaces/strategy';
+import { MilestoneStrategyTypeFlexible } from './MilestoneStrategyTypeFlexible';
+import GeneralStrategy from 'component/feature/StrategyTypes/GeneralStrategy/GeneralStrategy';
+import UserWithIdStrategy from 'component/feature/StrategyTypes/UserWithIdStrategy/UserWithId';
+import DefaultStrategy from 'component/feature/StrategyTypes/DefaultStrategy/DefaultStrategy';
+
+interface IMilestoneStrategyTypeProps {
+    strategy: Omit<IReleasePlanMilestoneStrategy, 'milestoneId'>;
+    strategyDefinition?: IStrategy;
+    parameters: IReleasePlanMilestoneStrategy['parameters'];
+    updateParameter: (field: string, value: string) => void;
+    errors: IFormErrors;
+}
+export const MilestoneStrategyType = ({
+    strategy,
+    strategyDefinition,
+    parameters,
+    updateParameter,
+    errors,
+}: IMilestoneStrategyTypeProps) => {
+    if (!strategyDefinition) {
+        return null;
+    }
+
+    switch (strategy.name) {
+        case 'default':
+            return <DefaultStrategy strategyDefinition={strategyDefinition} />;
+        case 'flexibleRollout':
+            return (
+                <MilestoneStrategyTypeFlexible
+                    parameters={parameters}
+                    updateParameter={updateParameter}
+                    errors={errors}
+                    editable={true}
+                />
+            );
+        case 'userWithId':
+            return (
+                <UserWithIdStrategy
```

Comment:
This along with the other UIs (Default and General) are all imported from the feature strategies. Should we make our own here or is this good enough for now?

Reply:
This is probably good enough for now, let's come back to this later.